sklearn/gaussian_process/gpc.py

```diff
@@ -0,0 +1,180 @@
+
+
+class GaussianProcessClassification(BaseEstimator):
+    """ Gaussian process classification (GPC).
+
+    The implementation is based on Algorithm 3.1, 3.2, and 5.1 of
+    ``Gaussian Processes for Machine Learning'' (GPML) by Rasmussen and
+    Williams.
+
+    Internally, the Laplace approximation is used for approximating the
+    non-Gaussian posterior by a Gaussian.
+
+    Currently, the implementation is restricted to
+      * using the logistic link function
+      * binary classification
+    """
+
+    def __init__(self, kernel, jitter=0.0):
+        self.kernel = kernel
+        self.jitter = jitter
+
+        # Values required for approximating the logistic sigmoid by
+        # error functions. coefs are obtained via:
+        # x = np.array([0, 0.6, 2, 3.5, 4.5, np.inf])
+        # b = logistic(x)
+        # A = (erf(np.dot(x, self.lambdas)) + 1) / 2
+        # coefs = lstsq(A, b)[0]
+        self.lambdas = np.array([0.41, 0.4, 0.37, 0.44, 0.39])[:, None]
+        self.coefs = np.array([-1854.8214151, 3516.89893646, 221.29346712,
+                               128.12323805, -2010.49422654])[:, None]
+
+    def fit(self, X, y):
+        # XXX: Assert that y is binary and labels are {0, 1}
+        self.X_fit_ = np.asarray(X)
+        self.y_fit_ = np.asarray(y)
+
+        if self.kernel.has_bounds:
+            # Choose hyperparameters based on maximizing the log-marginal
+            # likelihood
+            def obj_func(theta):
+                lml, grad = self.log_marginal_likelihood(theta,
+                                                         eval_gradient=True)
+                return -lml, -grad
+            self.theta_, lml, _ = fmin_l_bfgs_b(obj_func, self.kernel.params,
+                                                bounds=self.kernel.bounds)
+            self.kernel.params = self.theta_
+        else:
+            self.theta_ = self.kernel.params
+
+        # Precompute quantities required for predictions which are independent
+        # of actual query points
+        self.K_ = self.kernel.auto(self.X_fit_)
+        self.K_[np.diag_indices_from(self.K_)] += self.jitter
+
+        self.f_, _, (self.pi, self.W_sr, self.L, _, _) = \
+            self._posterior_mode(self.K_, return_temporaries=True)
+
+        return self
+
+    def predict(self, X):
+        # As discussed on Sectio 3.4.2 of GPML, for making hard binary
+        # decisions, it is enough to compute the MAP of the posterior and
+        # pass it through the link function
+        K_star = \
+            self.kernel.cross(self.X_fit_, X)  # K_star =k(x_star)
+        f_star = K_star.T.dot(self.y_fit_ - self.pi)  # Line 4
+
+        return f_star > 0
+
+    def predict_proba(self, X):
+        # Based on Algorithm 3.2 of GPML
+        K_star = \
+            self.kernel.cross(self.X_fit_, X)  # K_star =k(x_star)
+        f_star = K_star.T.dot(self.y_fit_ - self.pi)  # Line 4
+        v = solve(self.L, self.W_sr.dot(K_star))  # Line 5
```

Comment:
It could be helpful to turn these one letter variables into words. (Comment before reaching the end of the code, maybe I'll retract ;))